scan/connectivity_scanner.go

```diff
@@ -0,0 +1,57 @@
+package scan
+
+import "net"
+
+// ConnectivityScans contains scanners testing basic connectivity to the host
+var ConnectivityScanners = &Family{
+	Name:        "ConnectivityScanners",
+	Description: "Check for basic connectivity with the host",
+	Scanners: []*Scanner{
+		DNSLookupScanner,
+		TCPDialScanner,
+		TLSDialScanner,
+	},
+}
+
+// DNSLookupScanner tests that DNS resolution of the host returns at least one address
+var DNSLookupScanner = &Scanner{
+	"dnslookup",
+	"Host can be resolved through DNS",
+	dnsLookupScan,
+}
+
+func dnsLookupScan(host string) (grade Grade, output Output, err error) {
+	var conn net.Conn
+	addrs, err := net.LookupHost(host)
+	if err != nil || len(addrs) == 0 {
+		output = []string{}
+		return
+	}
+	defer conn.Close()
```

Comment:
This shouldn't be deferred; the conn is no longer in use and should be closed right away.